test/source/tests/setup.ts

```diff
@@ -2525,18 +2526,25 @@ AN8G3r5Htj8olot+jm9mIa5XLXWzMNUZgg==
     test(
       'setup - check custom authentication config from the local store (customer url fes)',
       testWithBrowser(async (t, browser) => {
+        const port = t.context.urls?.port ?? '';
         const oauthConfig = {
-          clientId: 'your-client-id',
-          clientSecret: 'your-client-secret',
-          redirectUrl: 'https://example.com/redirect',
-          authCodeUrl: 'https://example.com/auth',
-          tokensUrl: 'https://example.com/tokens',
+          clientId: OauthMock.customIDPClientId,
+          clientSecret: OauthMock.customIDPClientId,
+          redirectUrl: `https://google.localhost:${port}/robots.txt`,
```

Comment:
Okay, there is a bit of a difficulty here. We used `robots.txt` for Google IdP because it's the same org, so sharing auth result with them is safe.

But robots.txt should not be used for real authentication of 3rd party IdPs. I know this is just a mock but we should adjust it closer to what we expect it to be in prod.

What happens with real Auth0? Does it get sent to robots too? It should have its own redirect url provided by Auth0. When we use it, what happens? Is it a problem that our content script doesn't run there, or does it work fine?

Reply:
Well, I (IDP providers should also) set redirect_url to robots.txt in auth0 configuration.
Was not sure which redirect url to set and thought it would be safe to set redirect_url to robots.txt.
Maybe you prefer to set redirect url to our chrome extension url?(like chrome-extension://{id}/auth.js or something like that)?

Reply:
When customers set their custom redirect_url, it's challanging to get authorization code from this redirect_url as we can't inject content script in every site

Reply:
> When customers set their custom redirect_url, it's challanging to get authorization code from this redirect_url as we can't inject content script in every site

Right

> Maybe you prefer to set redirect url to our chrome extension url?(like chrome-extension://{id}/auth.js or something like that)?

IIRC Google IdP didn't allow this, causing us to use the hacky method with robots.txt. 

If Auth0 allows this then absolutely yes, we should use this method. Please see if it works as expected. The file could be called `oauth-redirect-page.htm` or similar, unless we already have such page, in which case we can reuse whatever we already have.

Reply:
> If we set the redirect URL to oauth-redirect-page.htm, it shows a page blocked error. I think we need to add chrome-extension://{id} to host-permissions in manifest.json which I think will trigger permission warning.

You're probably trying to set up content scripts at  oauth-redirect-page.htm which doesn't make sense. It's an extension page, you can just add `<script>` tags like any other extension page. See `attachment.htm` page or similar.

> Also, will the client be happy to set the redirect URL like chrome-extension://{id}? Maybe they prefer to set their own redirect_url?

No problem here, they'll set whatever we tell them to set, if it points to the installed app (addon or mobile app) and not to a web page we control.

Reply:
No I didn't. Just set redirect url to `oauth-redirect-page.htm` and after user finishes custom IDP authentication, it redirects to this page and above is result. (ERR_BLOCKED_BY_CLIENT)

>You're probably trying to set up content scripts at oauth-redirect-page.htm which doesn't make sense. It's an extension page, you can just add <script> tags like any other extension page. See attachment.htm page or similar.

Reply:
Maybe this one might be correct? It's chatgpt answer though.

> Chrome Security Policies: Chrome has strict security policies that might block certain behaviors that seem unusual or insecure, such as redirecting to a chrome-extension:// URL from a web context, even if it's part of an OAuth flow.